### PR TITLE
Implement User's personal image Gallery 

### PR DESCRIPTION
**Pull request by ethyus**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    
+    // Only query for users' images if logged in
+    List<Image> images = new ArrayList<>();
+    if (userService.isUserLoggedIn()) {
+        String userEmail = userService.getCurrentUser().getEmail();
+        
+        Query query = new Query("Image").setFilter(new FilterPredicate("email", FilterOperator.EQUAL, userEmail)).addSort("timestamp", SortDirection.DESCENDING);
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java:59]: Run java-format.

File: portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        }
+
+        usersImages.put("userImages", images);
+        String json = convertToJsonUsingGson(usersImages);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+  }
+  
+  // users post their own images
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java:82]: Users

File: portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      response.sendRedirect("/imageGallery.html");
+      return;
+    }
+
+    String blobKey = getBlobKey(request, "image");
+    String email = userService.getCurrentUser().getEmail();
+
+    if (blobKey == null) {
+    out.println("Please upload an image file.");
+    return;
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java:99]: Format this whole file.

File: portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java (hunk 2 above)
Comment (ethyus) [reply portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java]: done.

File: portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java (hunk 1 above)
Comment (ethyus) [reply portfolio/src/main/java/com/google/sps/servlets/NewImageDataServlet.java]: done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -79,28 +76,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-      // Send invalid input to client for correction
-      if (((Boolean) information.get("nameInfo").get("error") == true) || 
-          ((Boolean) information.get("messageInfo").get("error") == true)){
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    information = checkValidity(request);
 
-      } else {
+    // Send invalid input to client for correction
+    if (((Boolean) information.get("nameInfo").get("error") == true)
+        || ((Boolean) information.get("messageInfo").get("error") == true)) {
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:84]: Duplicate code from #8 .

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -111,33 +105,32 @@ private void storeComment(String kind, String name, String message){
...
-        name.put("error", false);
+      name.put("error", false);
     }
-    if (validateMessage(messageStr)){
-        message.put("error", true);
+    if (validateMessage(messageStr)) {
+      message.put("error", true);
     } else {
-        message.put("error", false);
+      message.put("error", false);
```
Comment (tylerhou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:133]: Duplicate code from #7.

File: portfolio/src/main/webapp/imageGallery.html [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width = device-width, initial-scale=1.0" />
+    <title>My Portfolio</title>
+    <link rel="stylesheet" href="style.css" />
+    <script src="script.js"></script>
+  </head>
+  <body onload = "startImageGallery()">
+      <div class = "global-container">
```
Comment (tylerhou) [inline portfolio/src/main/webapp/imageGallery.html:11]: No spaces around = in HTML

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -12,6 +12,109 @@
...
+    const imageWrapper = document.createElement('div');
+    imageWrapper.classList.add('img-wrap');
+
+    // Get Blob image from servlet
+    const imgElement = document.createElement('img');
+    imgElement.setAttribute('src', "/serve?key=" + image["blobKey"]);
+    imgElement.classList.add("userImage");
+
+    // Add delete button
+
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js:96]: Delete newline.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -12,6 +12,109 @@
...
+    });
+}
+
+function fetchUserStatus() {
+  fetch('/loginForGallery')
+    .then((response) => response.json())
+    .then((userJson) => {
+        const status = document.getElementById('logStatus');
+        const formStatus = document.getElementById('image-form');
+        if (userJson['email'] == 'null') {
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js:46]: Triple equals for string comparison.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -12,6 +12,109 @@
...
+    .then((response) => response.text())
+    .then((imageUploadUrl) => {
+      const messageForm = document.getElementById('my-form-gallery');
+      messageForm.action = imageUploadUrl;
+      messageForm.classList.remove('hidden');
+    });
+}
+
+function getImages() {
+
```
Comment (tylerhou) [inline portfolio/src/main/webapp/script.js:70]: Delete newline

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -124,6 +124,32 @@ h2 {
...
+
+.img-wrap .close {
+  position: absolute;
+  top: 2px;
+  right: 2px;
+  z-index: 100;
+  opacity: 0.1
+}
+
+.img-wrap:hover .close{
```
Comment (tylerhou) [inline portfolio/src/main/webapp/style.css:149]: Formatting.

File: portfolio/src/main/webapp/style.css [hunk 11]
```diff
@@ -205,6 +231,42 @@ h2 {
     padding: 0 0 10 0;
 }
 
+.section-5{
+    height: 600px;
+}
+
+.form-content{
+    padding: 20px 0 0 0;
+}
```
Comment (tylerhou) [inline portfolio/src/main/webapp/style.css:240]: Formatting.

**Review by tylerhou — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/style.css (hunk 10 above)
Comment (ethyus) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 11 above)
Comment (ethyus) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (ethyus) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (ethyus) [reply portfolio/src/main/webapp/script.js:94]: done

File: portfolio/src/main/webapp/imageGallery.html (hunk 6 above)
Comment (ethyus) [reply portfolio/src/main/webapp/imageGallery.html]: done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (ethyus) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: done

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (ethyus) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (ethyus) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:84]: done

Comment (tylerhou) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:84]: Was this done? I still see == true. Will #8 fix this?

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (tylerhou) [reply portfolio/src/main/webapp/script.js:94]: I still see a newline?

**Review by tylerhou — APPROVED** [review]

**Pull request merged**